### DISCUSSION TITLE
Latest official cli releases are running on Java 17 instead of gradle builded libs

### DISCUSSION THREAD

**Discussion by signals-from-outer-space** [AsamK/signal-cli#1405] [Q&A] [answered]
Hi,

with commit #80c1a6d the gradle build process was configured/updated for Java 21. It’s no longer possible to build (./gradlew build ...) or run the builded result with Java 17. 

Interestingly the latest releases on github are running on Java 17 w/o  any problems. What’s the difference in build process? How can i build Java 17 compatible libs?

Thanks a lot!

**Comment by AsamK** [comment, marked as answer]
The current master branch requires java 21.
The latest Bugfix releases are based on an older branch which still uses java 17.
The next bigger signal-cli release (0.13) will require java 21.
